packages/react-virtual/src/index.tsx

```diff
@@ -38,6 +38,7 @@ function useVirtualizerBase<
   )
 
   instance.setOptions(resolvedOptions)
+  instance.calculateRange()
```

Comment:
With this change the list renders the appropriate items when changing `options.count` in a single render:

https://user-images.githubusercontent.com/2291025/214837437-5a7815fd-da62-4658-8343-7e47d1057783.mov

But I am not sure if that's the best way to fix it for two reasons:

1. Calling `instance.setOptions` and/or `instance.calculateRange` in the body of the component seems risky in regards to React's concurrent mode. React can start a render cycle, which in this case would update the virtual instance's options and data, and abort that cycle. If that happens, the virtual instance would have been updated despite the abort, leading to mismatches between React's state and virtual's. I could be wrong though but that's how I understand it. The fix here would be to wrap them in a `useEffect`/`useLayoutEffect` (it'd be working slightly differently though by delaying updates).
2. I wonder if calling `calculateRange` every time the component renders wouldn't defeat other optimizations. From what I understand most things are memoized anyway but perhaps that wouldn't be ideal.